docs/integrations/mypy.md

```diff
@@ -46,162 +47,159 @@ But [with the plugin enabled](#enabling-the-plugin), it gives the correct error:
 
-For your code to pass with `--strict-optional`, you need to use `Optional[]` or an alias of `Optional[]`
-for all fields with `None` as the default. (This is standard with mypy.)
+## Supported mypy versions
 
-### Other Pydantic interfaces
+Pydantic supports the mypy versions released less than 6 months ago. Older versions may still work with the plugin
+but won't be tested. The list of released mypy versions can be found [here](https://mypy-lang.org/news.html). Note
+that the support policy is subject to changes if this becomes too much of an issue on our end.
 
-Pydantic [dataclasses](../concepts/dataclasses.md) and the [`validate_call` decorator](../concepts/validation_decorator.md)
-should also work well with mypy.
+## Mypy plugin capabilities
 
-## Mypy Plugin Capabilities
+### Generate a `__init__` signature for Pydantic models
 
-### Generate a signature for `Model.__init__`
 * Any required fields that don't have dynamically-determined aliases will be included as required
   keyword arguments.
-* If `Config.populate_by_name=True`, the generated signature will use the field names,
-  rather than aliases.
-* If `Config.extra='forbid'` and you don't make use of dynamically-determined aliases, the generated signature
-  will not allow unexpected inputs.
-* **Optional:** If the [`init_forbid_extra` **plugin setting**](#configuring-the-plugin) is set to `True`, unexpected inputs to
-  `__init__` will raise errors even if `Config.extra` is not `'forbid'`.
-* **Optional:** If the [`init_typed` **plugin setting**](#configuring-the-plugin) is set to `True`, the generated signature
-  will use the types of the model fields (otherwise they will be annotated as `Any` to allow parsing).
-
-### Generate a typed signature for `Model.model_construct`
-* The [`model_construct`](../concepts/models.md#creating-models-without-validation) method is an alternative to `__init__`
-  when input data is known to be valid and should not be parsed. Because this method performs no runtime validation,
-  static checking is important to detect errors.
-
-### Respect `Config.frozen`
-* If `Config.frozen` is `True`, you'll get a mypy error if you try to change
-  the value of a model field; cf. [faux immutability](../concepts/models.md#faux-immutability).
-
-### Generate a signature for `dataclasses`
-* classes decorated with [`@pydantic.dataclasses.dataclass`](../concepts/dataclasses.md) are type checked the same as standard Python dataclasses
-* The `@pydantic.dataclasses.dataclass` decorator accepts a `config` keyword argument which has the same meaning as [the `Config` sub-class](../concepts/config.md).
+* If the [`populate_by_name`][pydantic.ConfigDict.populate_by_name] model configuration value is set to
+  `True`, the generated signature will use the field names rather than aliases.
+* The [`init_forbid_extra`](#init_forbid_extra) and [`init_typed`](#init_typed) plugin configuration
+  values can further fine-tune the synthesized `__init__` method.
```

Comment:
I feel like the explicit specifications here were helpful - people often ask about `init_typed`...

Reply:
Then can just click on the `[init_typed](#init_typed)` reference, where I added a description for the setting (and same for others). Having things centralized avoids duplication, and having a heading makes it easier to link to when responding on github issues/discussions etc

To be clear, the explanation is not removed, just centralized in the configuration options section